File: colt-steele/countUniqueValues.js
// Attempt 1

function countUniqueValues(arr){
  let i = 0;
  let j = 1;
  
  if (arr.length === 0) {
      return 0;
  }
  
  while (j <= arr.length - 1) {
      if (arr[i] == arr[j]) {
          j++
      } else if (arr[i] !== arr[j]) {
          i++
          arr.splice(i, 1, arr[j])
          j++
      }
  }
  return i + 1
}

File: colt-steele/countUniqueValues.rb
def count_unique_values(arr)
    i = 0
    j = 1

    if arr.length === 0 
        puts 0
    end

    while j <= arr.length - 1
        if arr[i] == arr[j] 
            j += 1
        elsif arr[i] != arr[j]
            i++
            arr[i] = arr[j]
            j++
        end
    end

    puts i+1
end

